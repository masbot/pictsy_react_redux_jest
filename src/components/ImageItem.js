import React from 'react';

const ImageItem = (props) => (
    <div className="image-item">
        <img src={`/public/assets/images/${props.url}`} alt="cat pic" onClick={() => props.onClickImage(props.image)}/>
    </div>
)

export default ImageItem;