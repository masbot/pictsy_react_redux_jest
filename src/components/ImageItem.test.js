import React from 'react';
import { shallow } from 'enzyme';
import ImageItem from './ImageItem';

describe('ImageItem', () => {
    const mockOnClickImage = jest.fn();
    const props = { onClickImage: mockOnClickImage };
    const imageitem = shallow(<ImageItem {...props}/>);

    it('renders properly', () => {
        expect(imageitem).toMatchSnapshot();
    });

    describe('when user clicks on image', () => {
        beforeEach(() => {
            imageitem.find('img').simulate('click');
        });

        it('calls the onClickImage callback', () => {
            expect(mockOnClickImage).toHaveBeenCalled();
        });
    })
});
