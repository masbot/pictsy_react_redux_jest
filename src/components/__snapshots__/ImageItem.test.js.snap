// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageItem renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ImageItem
    onClickImage={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        alt="cat pic"
        onClick={[Function]}
        src="/public/assets/images/undefined"
/>,
      "className": "image-item",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "cat pic",
        "onClick": [Function],
        "src": "/public/assets/images/undefined",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="cat pic"
          onClick={[Function]}
          src="/public/assets/images/undefined"
/>,
        "className": "image-item",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "cat pic",
          "onClick": [Function],
          "src": "/public/assets/images/undefined",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
